src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/TypedArray.cs

```diff
@@ -51,31 +51,31 @@ public enum TypedArrayTypeCode
     /// </summary>
     public abstract class TypedArray<T, U> : CoreObject, ITypedArray, ITypedArray<T, U> where U : struct
     {
-        protected TypedArray() : base(Interop.Runtime.New<T>())
+        protected TypedArray() : base(typeof(T).Name)
```

Comment:
I think there's some bad ambiguity here since we only use .Name, someone could pass in a custom type with a name collision and get weird results (```Foo.Bar.Int32``` would just become ```Int32```), so we should use FullName

Reply:
As it is, this is looking up the JS host/core constructors in the `globalThis` which match the name and have no namespace. We could return to this question when we do custom types marshaling in 7.0 and introduce another lookup table instead of globalThis